x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilder.java

```diff
@@ -109,6 +182,101 @@ public void build() throws IOException {
         controller.startProcess(command);
     }
 
+    /**
+     * Visible for testing
+     */
+    List<String> buildAutodetectCommand() {
+        List<String> command = new ArrayList<>();
+        command.add(AUTODETECT_PATH);
+
+        String jobId = JOB_ID_ARG + job.getId();
+        command.add(jobId);
+
+        AnalysisConfig analysisConfig = job.getAnalysisConfig();
+        if (analysisConfig != null) {
+            addIfNotNull(analysisConfig.getBucketSpan(), BUCKET_SPAN_ARG, command);
+            addIfNotNull(analysisConfig.getLatency(), LATENCY_ARG, command);
+            addIfNotNull(analysisConfig.getSummaryCountFieldName(),
+                    SUMMARY_COUNT_FIELD_ARG, command);
+            if (Boolean.TRUE.equals(analysisConfig.getOverlappingBuckets())) {
+                Long window = analysisConfig.getResultFinalizationWindow();
+                if (window == null) {
+                    window = AnalysisConfig.DEFAULT_RESULT_FINALIZATION_WINDOW;
+                }
+                command.add(RESULT_FINALIZATION_WINDOW_ARG + window);
+            }
+            if (Boolean.TRUE.equals(analysisConfig.getMultivariateByFields())) {
+                command.add(MULTIVARIATE_BY_FIELDS_ARG);
+            }
+
+            if (analysisConfig.getUsePerPartitionNormalization()) {
+                command.add(PER_PARTITION_NORMALIZATION);
+            }
+        }
+
+        // Input is always length encoded
+        command.add(LENGTH_ENCODED_INPUT_ARG);
+
+        // Limit the number of output records
+        command.add(maxAnomalyRecordsArg(settings));
+
+        // always set the time field
+        String timeFieldArg = TIME_FIELD_ARG + getTimeFieldOrDefault(job);
+        command.add(timeFieldArg);
+
+        int intervalStagger = calculateStaggeringInterval(job.getId());
+        logger.debug("Periodic operations staggered by " + intervalStagger +" seconds for job '" + job.getId() + "'");
```

Comment:
I know you've only moved this message between files but it might be worth making the format more consistent with others, i.e. `[job_id] message`.